File: dotnet/Trinsic/TrustRegistryService.cs
using System.Threading.Tasks;
using Trinsic.Sdk.Options.V1;
using Trinsic.Services.TrustRegistry.V1;

namespace Trinsic;

public class TrustRegistryService : ServiceBase
{
    public TrustRegistryService(TrinsicOptions options)
        : base(options) {
        Client = new(Channel);
    }

    private TrustRegistry.TrustRegistryClient Client { get; }

    /// <summary>
    /// Search the trust registry
    /// </summary>
    /// <returns></returns>
    public async Task<SearchRegistryResponse> SearchAsync(SearchRegistryRequest request) {
        if (string.IsNullOrWhiteSpace(request.Query))
        {
            request.Query = "SELECT * FROM _ OFFSET 0 LIMIT 100";
        }

        var response = await SearchRegistryAsync(request);
        return response;
    }

    public SearchRegistryResponse Search(SearchRegistryRequest request) {
        if (string.IsNullOrWhiteSpace(request.Query))
        {
            request.Query = "SELECT * FROM _ OFFSET 0 LIMIT 100";
        }

        return SearchRegistry(request);
    }
// BEGIN Code generated by protoc-gen-trinsic. DO NOT EDIT.
// target: /home/runner/work/sdk/sdk/dotnet/Trinsic/TrustRegistryService.cs

	/// <summary>
    /// Add a governance framework to the ecosystem
    /// </summary>
    
    public AddFrameworkResponse AddFramework(AddFrameworkRequest request) {
        
        return Client.AddFramework(request, BuildMetadata(request));
    }
	
	/// <summary>
    /// Add a governance framework to the ecosystem
    /// </summary>
    
    public async Task<AddFrameworkResponse> AddFrameworkAsync(AddFrameworkRequest request) {
        
        return await Client.AddFrameworkAsync(request, await BuildMetadataAsync(request));
    }

	/// <summary>
    /// Remove a governance framework from the ecosystem
    /// </summary>
    
    public RemoveFrameworkResponse RemoveFramework(RemoveFrameworkRequest request) {
        
        return Client.RemoveFramework(request, BuildMetadata(request));
    }
	
	/// <summary>
    /// Remove a governance framework from the ecosystem
    /// </summary>
    
    public async Task<RemoveFrameworkResponse> RemoveFrameworkAsync(RemoveFrameworkRequest request) {
        
        return await Client.RemoveFrameworkAsync(request, await BuildMetadataAsync(request));
    }

	/// <summary>
    /// Search the ecosystem's governance frameworks
    /// </summary>
    
    public SearchRegistryResponse SearchRegistry(SearchRegistryRequest request) {
        
        return Client.SearchRegistry(request, BuildMetadata(request));
    }
	
	/// <summary>
    /// Search the ecosystem's governance frameworks
    /// </summary>
    
    public async Task<SearchRegistryResponse> SearchRegistryAsync(SearchRegistryRequest request) {
        
        return await Client.SearchRegistryAsync(request, await BuildMetadataAsync(request));
    }

	/// <summary>
    /// Register an authoritative issuer for a credential schema
    /// </summary>
    
    public RegisterMemberResponse RegisterMember(RegisterMemberRequest request) {
        
        return Client.RegisterMember(request, BuildMetadata(request));
    }
	
	/// <summary>
    /// Register an authoritative issuer for a credential schema
    /// </summary>
    
    public async Task<RegisterMemberResponse> RegisterMemberAsync(RegisterMemberRequest request) {
        
        return await Client.RegisterMemberAsync(request, await BuildMetadataAsync(request));
    }

	/// <summary>
    /// Removes an authoritative issuer for a credential schema from the trust registry
    /// </summary>
    
    public UnregisterMemberResponse UnregisterMember(UnregisterMemberRequest request) {
        
        return Client.UnregisterMember(request, BuildMetadata(request));
    }
	
	/// <summary>
    /// Removes an authoritative issuer for a credential schema from the trust registry
    /// </summary>
    
    public async Task<UnregisterMemberResponse> UnregisterMemberAsync(UnregisterMemberRequest request) {
        
        return await Client.UnregisterMemberAsync(request, await BuildMetadataAsync(request));
    }

	/// <summary>
    /// Fetch the membership status of an issuer for a given credential schema in a trust registry
    /// </summary>
    
    public GetMembershipStatusResponse GetMembershipStatus(GetMembershipStatusRequest request) {
        
        return Client.GetMembershipStatus(request, BuildMetadata(request));
    }
	
	/// <summary>
    /// Fetch the membership status of an issuer for a given credential schema in a trust registry
    /// </summary>
    
    public async Task<GetMembershipStatusResponse> GetMembershipStatusAsync(GetMembershipStatusRequest request) {
        
        return await Client.GetMembershipStatusAsync(request, await BuildMetadataAsync(request));
    }

	/// <summary>
    /// Fetch the ecosystem's authorized issuers and the respective templates against which it can issue
    /// </summary>
    
    public ListAuthorizedMembersResponse ListAuthorizedMembers(ListAuthorizedMembersRequest request) {
        
        return Client.ListAuthorizedMembers(request, BuildMetadata(request));
    }
	
	/// <summary>
    /// Fetch the ecosystem's authorized issuers and the respective templates against which it can issue
    /// </summary>
    
    public async Task<ListAuthorizedMembersResponse> ListAuthorizedMembersAsync(ListAuthorizedMembersRequest request) {
        
        return await Client.ListAuthorizedMembersAsync(request, await BuildMetadataAsync(request));
    }

// END Code generated by protoc-gen-trinsic. DO NOT EDIT.
}

File: python/trinsic/trustregistry_service.py
from trinsic.proto.sdk.options.v1 import ServiceOptions
from trinsic.proto.services.trustregistry.v1 import *
from trinsic.security_providers import ITokenProvider
from trinsic.service_base import ServiceBase


class TrustRegistryService(ServiceBase):
    """
    Wrapper for [Trust Registry Service](/reference/services/trust-registry/)
    """

    def __init__(
        self,
        *,
        server_config: ServiceOptions = None,
        token_provider: ITokenProvider = None
    ):
        super().__init__(server_config, token_provider)
        self.client = TrustRegistryStub(super().channel)

    async def search(
        self, *, request: SearchRegistryRequest = None
    ) -> SearchRegistryResponse:
        """
        [Search the registry](/reference/services/trust-registry/#search)
        Args:
        Returns:
            [SearchRegistryResponse](/reference/proto/#searchregistryresponse)
        """
        request = request or SearchRegistryRequest()
        request.query = request.query or "SELECT * FROM c OFFSET 0 LIMIT 100"
        return await self.search_registry(request=request)

# BEGIN Code generated by protoc-gen-trinsic. DO NOT EDIT.
# target: C:\work\sdk\python\trinsic\trustregistry_service.py

    async def add_framework(self, *, request: AddFrameworkRequest) -> AddFrameworkResponse:
        """  Add a governance framework to the ecosystem """
        
        return await self.client.add_framework(request, metadata=self.build_metadata(request))
    
    async def remove_framework(self, *, request: RemoveFrameworkRequest) -> RemoveFrameworkResponse:
        """  Remove a governance framework from the ecosystem """
        
        return await self.client.remove_framework(request, metadata=self.build_metadata(request))
    
    async def search_registry(self, *, request: SearchRegistryRequest) -> SearchRegistryResponse:
        """  Search the ecosystem's governance frameworks """
        
        return await self.client.search_registry(request, metadata=self.build_metadata(request))
    
    async def register_member(self, *, request: RegisterMemberRequest) -> RegisterMemberResponse:
        """  Register an authoritative issuer for a credential schema """
        
        return await self.client.register_member(request, metadata=self.build_metadata(request))
    
    async def unregister_member(self, *, request: UnregisterMemberRequest) -> UnregisterMemberResponse:
        """  Removes an authoritative issuer for a credential schema from the trust registry """
        
        return await self.client.unregister_member(request, metadata=self.build_metadata(request))
    
    async def get_membership_status(self, *, request: GetMembershipStatusRequest) -> GetMembershipStatusResponse:
        """  Fetch the membership status of an issuer for a given credential schema in a trust registry """
        
        return await self.client.get_membership_status(request, metadata=self.build_metadata(request))
    

# END Code generated by protoc-gen-trinsic. DO NOT EDIT.
